core/controllers/acl_decorators.py

```diff
@@ -1927,6 +1927,39 @@ def test_can_publish_skill(self, skill_id, **kwargs):
     return test_can_publish_skill
 
 
+def can_access_user_skill_mastery(handler):
```

Comment:
All this decorator does is checks if the user has logged in. I wonder if we want to make this a generic decorator rather than specific to skill mastery. Or do we already have a decorator that is meant to only check if user is logged in? I am not very aware of decorators, @seanlip can you help here?

Reply:
I don't think there are any that does that, though `require_user_id_else_redirect_to_homepage` does check only  logged in status and redirects if not.

Reply:
Should we make a generic decorator for is_logged_in or something?

Reply:
Could you just use something functionally similar instead, e.g. can_access_learner_dashboard (which, in general, keeps track of the user's progress)? I think that would be the way to go. I agree with not proliferating decorators if it is not necessary to do so.

Reply:
Done, but will it cause any confusion? Since the name doesn't match.

Reply:
I think it's fine -- though maybe it might be better to rename the decorator to something like can_view_lesson_progress or something like that?

(Also, otherwise, each controller has its own acl decorator and that's what we are trying to avoid.)

Reply:
Actually, when I looked into it, I feel like there is no need to change its name. This decorator is also used in many other situations, and it would be hard to generalize all those functionalities. And the logic here makes sense - as long as you can view the first page (learner dashboard), you can do these things as well.

Reply:
OK, LGTM. Thanks!